pkg/k8s/util.go

```diff
@@ -23,29 +49,111 @@ func PodIsStatic(pod *v1.Pod) bool {
 	return ok && configSource == "file"
 }
 
-// CalculatePodsRequestsTotal returns the total capacity of all pods
-func CalculatePodsRequestsTotal(pods []*v1.Pod) (resource.Quantity, resource.Quantity, error) {
-	memoryRequests := *k8s_resource.NewMemoryQuantity(0)
-	cpuRequests := *k8s_resource.NewCPUQuantity(0)
+// CalculatePodsRequestedUsage returns the total capacity of all pods
+func CalculatePodsRequestedUsage(pods []*v1.Pod) (PodRequestedUsage, error) {
+	ret := newPodRequestedUsage()
 
 	for _, pod := range pods {
 		podResources := scheduler.ComputePodResourceRequest(pod)
-		memoryRequests.Add(*k8s_resource.NewMemoryQuantity(podResources.Memory))
-		cpuRequests.Add(*k8s_resource.NewCPUQuantity(podResources.MilliCPU))
+		ret.Total.Memory += podResources.Memory
+		ret.Total.MilliCPU += podResources.MilliCPU
+		if pod.Status.Phase == v1.PodPending {
+			if podResources.Memory > ret.LargestMemory.Memory {
+				ret.LargestMemory = scheduler.NewResource(podResources.MilliCPU, podResources.Memory)
```

Comment:
What happens if we have no pods pending and therefore LargestMemory/LargestCPU are zero?